vale/styles/jargon/TechnologySpecials.yml

```
@@ -0,0 +1,40 @@
+---
+extends: substitution
+message: Use '%s' instead of '%s'
+level: warning
+ignorecase: true
+scope: raw
+swap:
+  # negative look behind was not enough (?<![\./])
+  # so only positive look behind was used (?<=\s)
+  #
+  # technology but also file extension
+  # manualcherry picking in
+  # https://en.wikipedia.org/wiki/List_of_filename_extensions
+  # https://en.wikipedia.org/wiki/List_of_file_formats
+  (?<=\s)CSON: CSON
```

```suggestion
  \bCSON\b: CSON
```

I suggest using word break `\b` instead of looking for a space. There might not be a space.

---

Please note: Look behind is not supported by all browsers. Safari only added it last year: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Lookbehind_assertion

---

> ```suggestion
>   \bCSON\b: CSON
> ```
> 
> I suggest using word break `\b` instead of looking for a space. There might not be a space.
> 

Great remark I will have a look and check for the use cases I added this pattern for things that could be extensions and terms

But I'm sure it was about handing such use case:

```
JSON is a format, the related extension is .json, example whatever.json 
```

I don't want Vale to report an issue about this valid text.

---

> Please note: Look behind is not supported by all browsers. Safari only added it last year: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Lookbehind_assertion

Indeed, but this is a for for vale (a validation tool)

It's in Go, and it supports the syntax.

But I might to keep that in mind because there is a lsp server written in JavaScript for vale

---

I think the \b is implicit with Vale, unless I'm playing with `nonword:true`

nonword, bool, Removes the default word boundaries ( \b )

---

Anyway, the issue I had was with extension

so here is what I need

```
(?<!\.)CSON: CSON
```

a negative look behind